### PR TITLE
[stable/prometheus-operator] update doc for prometheus.prometheusSpec…

### LINKED ISSUE #9140: [stable/prometheus-operator] servicemonitor are not loaded
**Is this a request for help?**: Maybe

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**:

Helm : 2.11.0
K8s : 1.11.3
On Azure AKS

**Which chart**: stable/prometheus-operator

**What happened**:

I created a custom `ServiceMonitor` in my `monitoring` namespace where I installed the chart:

```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: monitoring-prometheus-oper-influxdb
  namespace: monitoring
spec:
  jobLabel: influxdb
  selector:
    matchLabels:
      app: influxdb
  namespaceSelector:
    matchNames:
      - prod
  endpoints:
    - port: exporter
      interval: 30s
```

However it is not pick up by Prometheus. After more tant 20 minutes, I still don't see the target on Prometheus web UI.

**What you expected to happen**:

The target to be created.

**How to reproduce it** (as minimally and precisely as possible):

1. Create an AKS cluster
2. Install the chart with standard values but with rbac disabled
3. Create any custom servicemonitor resources. 

It is not picked up.

**Anything else we need to know**: None

### PR DESCRIPTION
**Pull request by jbouzekri**
….serviceMonitorSelector value

#### What this PR does / why we need it:

It fixes a documentation issue about how the `prometheus.prometheusSpec.serviceMonitorSelector` value behaves in the chart stable/prometheus-operator

#### Which issue this PR fixes

  - fixes #9140 

#### Checklist

- [ ] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

**Comment by vsliouniaev** [comment]
@jbouzekri thanks for this, but you need to bump the chart version and add sign-off to the commits for CI to pass

**Review by gianrubio — COMMENTED** [review]
/lgtm /ok-to-test

**Comment by paskal** [comment]
I think you need to merge upstream/master in order to proceed with merge, @jbouzekri

**Comment by paskal** [comment]
@jbouzekri, I think you need to do it once more, another one got merged.

**Comment by jbouzekri** [comment]
@paskal @gianrubio @vsliouniaev : done third time

**Comment by paskal** [comment]
@gianrubio, could you please accept this once more?

**Comment by paskal** [comment]
Damn race conditions... could you please bump once more? @gianrubio, could you please merge it finally after that?

**Comment by tim-beatson** [comment]
Whilst the documentation at present is incorrect, and this PR updates the docs to reflect accurately what the chart does, I have reservations about the way this is implemented.

First of all, I think it's bad practice to ship a helm chart that renders differently to what is in the default values file, but at present this is the case.

Secondly, what if I don't want to use serviceMonitorSelector? My use case is one Prometheus instance per k8s cluster, so Prom should be realising all service monitors. The current implementation makes this impossible.

I would suggest removing the else statement entirely and have the values file and either place what is under the else statement there, or (this would be my preference) just leave the values file as is. Certainly the else statement should be removed as it's miss leading to have the template go away and render something that differs from the default values.

See https://github.com/helm/charts/issues/9621

**Comment by tim-beatson** [comment]
@paskal, see comment above as requested. TL;DR, I'd rather the default be that all are loaded.

**Comment by jbouzekri** [comment]
@paskal : done

**Comment by paskal** [comment]
@gianrubio @davidkarlsen could someone please re-approve this simple change?

**Review by gianrubio — APPROVED** [review]
/ok-to-test
/lgtm

**Comment by paskal** [comment]
@jbouzekri could you please rebase is on upstream\master and bump version once more? Everything more important is already merged, and there is a very good chance this will be merged now finally.

**Pull request closed without merging**
